File: LeetCode Solutions/Problems/83. Remove Duplicates from Sorted List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if(head == null){
            return head;
        }
        ListNode temp = head.next;
        ListNode previous = head;
        while(temp != null){
            if(temp.val == previous.val){
                previous.next = temp.next;
            } else{
                previous = temp;
            }
            temp = temp.next;
        }
        return head;
    }
}

File: LeetCode Solutions/Problems/83. Remove Duplicates from Sorted List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
        if(not head):
            return head
        temp = head.next
        previous = head
        while(temp):
            if(temp.val == previous.val):
                previous.next = temp.next
            else:
                previous = temp
            temp = temp.next
        return head